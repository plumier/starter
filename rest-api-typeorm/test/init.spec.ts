import supertest from "supertest"
import createApp from "../src/app"
import "@plumier/testing"
import {join} from "path"
import dotenv from "dotenv"

dotenv.config({ path: join(__dirname, "./.env-test") })

it("Should initialized properly", async () => {
    const mock = console.mock()
    const app = await createApp()
    console.mockClear()
    expect(mock.mock.calls).toMatchSnapshot()
})
