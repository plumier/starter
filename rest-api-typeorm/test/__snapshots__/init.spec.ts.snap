// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should initialized properly 1`] = `
Array [
  Array [],
  Array [
    "Route Analysis Report",
  ],
  Array [
    " 1. AuthController.login(email, password)    -> Public        POST   /auth/login",
  ],
  Array [
    " 2. AuthController.refresh(user)             -> RefreshToken  POST   /auth/refresh",
  ],
  Array [
    " 3. AuthController.logout()                  -> AnyUser       POST   /auth/logout",
  ],
  Array [
    " 4. TypeORMControllerGeneric.list            -> Admin         GET    /users",
  ],
  Array [
    " 5. TypeORMControllerGeneric.save(data, ctx) -> Public        POST   /users",
  ],
  Array [
    " 6. TypeORMControllerGeneric.get             -> ResourceOwner GET    /users/:id",
  ],
  Array [
    " 7. TypeORMControllerGeneric.modify          -> ResourceOwner PATCH  /users/:id",
  ],
  Array [
    " 8. TypeORMControllerGeneric.replace         -> ResourceOwner PUT    /users/:id",
  ],
  Array [
    " 9. TypeORMControllerGeneric.delete(id, ctx) -> ResourceOwner DELETE /users/:id",
  ],
  Array [
    "10. SwaggerFacility                          -> Public        GET    /swagger",
  ],
  Array [],
]
`;
